### PR TITLE
Add interfacer and unconvert linters

### PR DESCRIPTION
**Pull request by dnephin**
`interfacer` checks function arguments for types that could use a smaller interface
`unconvert` finds unnecessary type convertions

### CODE REVIEW

**Comment by dmcgowan** [comment]
The 2 places where `// nolint: interfacer` is used a `string` could just be passed in instead. Would that be considered a functional change you are trying to avoid here?

**Comment by dnephin** [comment]
Yes, I considered making that change, but I decided against it (at least for now). There are cases (like these) where I think it's actually better to accept a more specific type, even if it's only being used as a string, because it gives the caller more information about the supported values.

`PingV2Registry` is exported as well, so I didn't want to break any external consumers of that package (docker/cli for example) when they go to upgrade.

File: api/server/router/container/backend.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ type execBackend interface {
 	ContainerExecCreate(name string, config *types.ExecConfig) (string, error)
 	ContainerExecInspect(id string) (*backend.ExecInspect, error)
 	ContainerExecResize(name string, height, width int) error
-	ContainerExecStart(ctx context.Context, name string, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) error
+	ContainerExecStart(ctx context.Context, name string, stdin io.Reader, stdout io.Writer, stderr io.Writer) error
```
Comment (cpuguy83) [inline api/server/router/container/backend.go]: This one seems a bit string.
I'd expect to need to close `stdin` if the exec/container process exits.

Comment (dnephin) [reply api/server/router/container/backend.go]: Yes, that's what I was thinking too, but apparently that's not the case.

I'll take another look to find out why.

Comment (dnephin) [reply api/server/router/container/backend.go]: It looks like the caller is responsible for closing the the reader, which is why this isn't using `ReadCloser`

**Comment by dnephin** [comment]
ping. Anything I should change here, or do my explanations make sense?

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by dmcgowan — APPROVED** [review]
LGTM

**Comment by Battleborn69** [comment]
Thank you kindly

**Review by yongtang — APPROVED** [review]
LGTM

**Comment by kolyshkin** [comment]
I am seeing ARM build failure because of commit 2f5f0af3f, here's the error:
```
19:06:45 ---> Making bundle: dynbinary (in bundles/17.06.0-dev/dynbinary)
19:06:48 Building: bundles/17.06.0-dev/dynbinary-daemon/dockerd-17.06.0-dev
19:10:58 # github.com/docker/docker/daemon/graphdriver/overlay
19:10:58 daemon/graphdriver/overlay/copy.go:161: cannot use stat.Atim.Sec (type int32) as type int64 in argument to time.Unix
19:10:58 daemon/graphdriver/overlay/copy.go:161: cannot use stat.Atim.Nsec (type int32) as type int64 in argument to time.Unix
19:10:58 daemon/graphdriver/overlay/copy.go:162: cannot use stat.Mtim.Sec (type int32) as type int64 in argument to time.Unix
19:10:58 daemon/graphdriver/overlay/copy.go:162: cannot use stat.Mtim.Nsec (type int32) as type int64 in argument to time.Unix
```

**Pull request merged**